### PR TITLE
roon-tui: init at 0.3.0

### PR DESCRIPTION
**Pull request by MichaelCDormann**
## Description of changes

[Roon Server](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/audio/roon-server.nix) and [Roon Bridge](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/audio/roon-bridge.nix) already exist in nixpkgs. Roon tui allows you to control your Roon devices from the terminal.

https://github.com/TheAppgineer/roon-tui

This is my first time contributing to nixpkgs. Let me know if I missed anything!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ro/roon-tui/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      "roon-api-0.1.1" = "sha256-aFcS8esfgMxzzhWLeynTRFp1FZj2z6aHIivU/5p+uec=";
+    };
+  };
+
+  meta = {
+    description = "A Roon Remote for the terminal";
+    homepage = "https://github.com/TheAppgineer/roon-tui";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ MichaelCDormann ];
+    platforms = [ "x86_64-linux" ];
```
Comment (lolbinarycat) [inline pkgs/by-name/ro/roon-tui/package.nix:29]: this could use a [meta.mainProgram](https://nixos.org/manual/nixpkgs/stable/#var-meta-mainProgram) key

Comment (MichaelCDormann) [reply pkgs/by-name/ro/roon-tui/package.nix:29]: Thanks, done!

File: pkgs/by-name/ro/roon-tui/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
```
Comment (SuperSandro2000) [inline pkgs/by-name/ro/roon-tui/package.nix:29]: Why shouldn't this work on other platforms?

Comment (MichaelCDormann) [reply pkgs/by-name/ro/roon-tui/package.nix:29]: From what I understand it probably could, but I can only test and verify on x86_64-linux. I figured if somebody wanted to verify that it works on other platforms and help maintain this derivation for those platforms they can. Is the best approach to go ahead and enable other platforms anyway?

Comment (AndersonTorres) [inline pkgs/by-name/ro/roon-tui/package.nix:29]: When you fix the platforms list this way, you are basically saying the upstream developers develop it for that specific platform only. A typical example is the `zsnes` emulator, written in x86 assembly.

I believe this is not the case here. Just remove this line:

```suggestion
```

Here around Nixpkgs we just wait for volunteers. Indeed a lot of people help keeping mpv and emacs on Darwin.

Until then, if ofborg fails then you can add something like

 ```nix
meta = { 
  . . . 
  broken = stdenv.isDarwin; # ofborg failed
}
```

File: pkgs/by-name/ro/roon-tui/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+, rustPlatform
+, fetchFromGitHub
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "roon-tui";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "TheAppgineer";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/by-name/ro/roon-tui/package.nix:12]:
```suggestion
    repo = "roon-tui";
```

https://github.com/NixOS/nixpkgs/issues/277994

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ro/roon-tui/package.nix (hunk 2 above)
Comment (MichaelCDormann) [reply pkgs/by-name/ro/roon-tui/package.nix]: Got it, that makes sense

**Review by AndersonTorres — COMMENTED** [review]
LGTM.

File: pkgs/by-name/ro/roon-tui/package.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/by-name/ro/roon-tui/package.nix]: The approach is to set this to all platforms that theoretical work, like all linux and then see what other users report and what the CI is telling us.

**Pull request merged**